sbin/functions.sh

```diff
@@ -964,6 +964,469 @@ clean_up()
+n
+p
+2
+
++`numfmt --to=iec --from=iec ${swappartsize}`
+n
+p
+3
+
++`numfmt --to=iec --from=iec ${rootfspartsize}`
+n
+p
+
++`numfmt --to=iec --from=iec ${containerpartsize}`
+a
+1
+t
+1
+b
+t
+2
+82
+t
+3
+83
+w
+p
+q
+
+
+EOF
+		echo "FDISK_PARTITION_LAYOUT=\"${localpartitionlayout}\"" >> ${localconf}
+	fi
+}
+
+interactive_config()
+{
+	local device="$1"
+	local localconf_dir="$2"
+
+	if [ -z ${device} ]
+	then
+		debugmsg ${DEBUG_CRIT} "ERROR in interactive_config: device parameter not provided"
+		return 1
+	fi
+
+	if [ -z ${localconf_dir} ]
+	then
+		debugmsg ${DEBUG_CRIT} "ERROR in interactive_config: tmp config directory parameter not provided"
+		return 1
+	fi
+
+	local localconf="${localconf_dir}/config.sh"
+	tmpfile=$(mktemp)
+
+	echo "" > ${localconf}
+
+	# Distribution
+	dialog --no-cancel --inputbox "Distribution name:" 8 40 "OverC" 2>$tmpfile
+	local dialog_res=$(cat $tmpfile)
+	if [ -n "${dialog_res}" ]; then
+		echo "DISTRIBUTION=\"${dialog_res}\"" >> ${localconf}
+	else
+		echo "DISTRIBUTION=\"OverC\"" >> ${localconf}
+	fi
+
+	# Container selection
+	local filecount=0
+	local filelist="" menuitems=""
+	if [ -z "$ARTIFACTS_DIR" ]; then
+		if [ -e "tmp/deploy/images" ]; then
+			filecount=`ls tmp/deploy/images/ -1 | wc -l`
+			case "$filecount" in
+				0)
+					debugmsg ${DEBUG_CRIT}  "Project build incomplete. Please try finish the build first."
+					return 1
+					;;
+				1)
+					ARTIFACTS_DIR="`pwd`/tmp/deploy/images/`ls tmp/deploy/images/`"
+					;;
+				*)
+					filelist=`ls tmp/deploy/images` menuitems=""
+					for i in $filelist; do
+						menuitems="$menuitems $i $i "
+					done
+					dialog --no-cancel --menu "Found multiple machine target in your project, Please choose one:" \
+						$(expr $filecount + 7) 30 $filecount $menuitems 2>$tmpfile
+					ARTIFACTS_DIR="`pwd`/tmp/deploy/images/$(cat $tmpfile)"
+					;;
+			esac
+		elif [ -e "images" ]; then
+			ARTIFACTS_DIR="`pwd`/images"
+		else
+			# default to current dir
+			ARTIFACTS_DIR="`pwd`"
+		fi
+	fi
+	echo "ARTIFACTS_DIR=\"${ARTIFACTS_DIR}\"" >> ${localconf}
+	filecount=`ls $ARTIFACTS_DIR -1 | grep -v menifest | wc -l`
+	if [ $filecount -lt 3 ]; then
+		debugmsg ${DEBUG_CRIT} "Artifacts are not sufficient. Please verify."
+		return 1
+	fi
+
+	# exclude manifest files
+	filelist=`ls $ARTIFACTS_DIR | grep -v manifest` menuitems=""
+	local checklistmenu=""
+	for i in $filelist; do
+		menuitems="$menuitems $i $i "
+		checklistmenu="$checklistmenu $i $i off "
+	done
+
+	# Kernel selection
+	dialog --no-tags --menu "Please choose kernel:" \
+		$(expr $filecount + 7) 100 $filecount $menuitems 2>$tmpfile
+	dialog_res=$(cat $tmpfile)
+	if [ -z "${dialog_res}" ]; then
+		debugmsg ${DEBUG_CRIT} "Kernel not specified. Please try again."
+		return 1
+	fi
+	echo "INSTALL_KERNEL=\"\${ARTIFACTS_DIR}/${dialog_res}\"" >> ${localconf}
+
+	# Rootfs selection
+	dialog --no-tags --menu "Please choose rootfs:" \
+		$(expr $filecount + 7) 100 $filecount $menuitems 2>$tmpfile
```

Comment:
If we restrict the name/rootfs file type, user might be confused if they made their own image format other than our restricted one. If you think we should do this restriction, please just let me know how  we should define the rootfs type.